networkml/trained_models/README.md

```diff
@@ -2,25 +2,55 @@
 
 ## Overview
 
-NetworkML performs role identification via supervised machine learning. Although our 
+NetworkML performs role identification via supervised machine learning. Although our
 internal analysis compared decision trees, random forests, and neural networks, the public
 networkML codebase only includes a neural network (or "deep learning") model.
 
 ### Neural Network
 Neural networks can be used for supervised machine learning to match patterns in network
 traffic with the functional role of a device. For further information on neural networks,
 see Francois Chollet's "Deep Learning with Python" published by Manning
-Publications in 2018, especially pages 3-116. For more general information on machine 
-learning and information security or cybersecurity, see Clarence 
+Publications in 2018, especially pages 3-116. For more general information on machine
+learning and information security or cybersecurity, see Clarence
 Chio and David Freeman, "Machine Learning & Security," published by O'Reilly
 in 2018. The neural network model in networkML uses the Python package scikit-learn. Using
 TensorFlow or a similar neural network-specific machine learning package was not necessary
 to achieve high levels of performance in our in-house testing.
 
 ### host_footprint model
 
-Currently one model is defined, depending upon the Host() featurizer. It was generated by:
+Currently one model is defined, depending upon the Host() featurizer.
+
+#### Training and predicting
+
+The model can be regenerated by:
 
 ~~~~
-networkml --first_stage=algorithm --final_stage=algorithm --trained_model=networkml/trained_models/host_footprint.json --label_encoder=networkml/trained_models/host_footprint_le.json --operation train --kfolds 5 ~/tmp/combined-host-ratios-041020.gz
+networkml --kfolds=5 --first_stage=algorithm --trained_model=networkml/trained_models/host_footprint.json --label_encoder=networkml/trained_models/host_footprint_le.json --scaler=networkml/trained_models/host_footprint_scaler.mod --operation train [--test_data=/tmp/test_host.csv] /tmp/train_host.csv
```

Comment:
Does this command (given the current codebase) still include the unix time-related statistical features? I've been thinking that we should perhaps remove those statistical features from the code base. I worry that unsuspecting users will include those, get unrealistically good results, and not understand that the unrealistically good results are an artifact of the unix time-related statistical features.

Reply:
No - class HostBase:

    CALC_COL_NAMES = (
        ('frame.len', 'frame_len'),
        ('frame.time_delta_displayed', 'time_delta'))

Only frame.len and frame.time_deta_displayed (interpacket arrival time). The canonical list is of course the featurizer CSV submitted to the training stage.

>>> import pandas as pd
>>> df = pd.read_csv('combined-host-ratios-041020.gz')
>>> cols = sorted(df.columns)
>>> [col for col in cols if 'time' in col]
['tshark_25q_time_delta_in', 'tshark_25q_time_delta_out', 'tshark_75q_time_delta_in', 'tshark_75q_time_delta_out', 'tshark_average_time_delta_in', 'tshark_average_time_delta_out', 'tshark_count_time_delta_in', 'tshark_count_time_delta_out', 'tshark_max_time_delta_in', 'tshark_max_time_delta_out', 'tshark_median_time_delta_in', 'tshark_median_time_delta_out', 'tshark_min_time_delta_in', 'tshark_min_time_delta_out', 'tshark_total_time_delta_in', 'tshark_total_time_delta_out', 'tshark_variance_time_delta_in', 'tshark_variance_time_delta_out'